import-export-cli/cmd/k8s/addApi.go

```diff
@@ -0,0 +1,230 @@
+available modes are as follows
+* kubernetes`
+const addApiExamples = utils.ProjectName + " " + K8sCmdLiteral + " add/update " + AddApiCmdLiteral +
+	` -n petstore --from-file=./Swagger.json --replicas=3 --namespace=wso2`
+
+// addApiCmd represents the api command
+var addApiCmd = &cobra.Command{
+	Use:     AddApiCmdLiteral,
+	Short:   addApiCmdShortDesc,
+	Long:    addApiLongDesc,
+	Example: addApiExamples,
+	Run: func(cmd *cobra.Command, args []string) {
+		utils.Logln(utils.LogPrefixInfo + AddApiCmdLiteral + " called")
+		handleAddApi("")
+	},
+}
+
+func handleAddApi(nameSuffix string) {
+	validateAddApiCommand()
+
+	//swaggerCmNames := make([]string, len(flagSwaggerFilePaths))
+	//balInterceptorsCmNames := make([]string, 0, len(flagSwaggerFilePaths))
+	//var javaInterceptorsCmNames []string
+
+	// log processing only if there are more projects
+	utils.Logln(fmt.Sprintf("%sProcessing swagger  %v", utils.LogPrefixInfo, flagSwaggerFilePath))
+
+	flagApiName = strings.ToLower(flagApiName)
+	swaggerCmName := fmt.Sprintf("%v-swagger%s", flagApiName, nameSuffix)
+
+	fi, _ := os.Stat(flagSwaggerFilePath) // error already handled and ignore error
+	switch mode := fi.Mode(); {
+	//check if the swagger path is a Dir
+	case mode.IsDir():
+		//get swagger definition
+		swaggerPath := filepath.Join(flagSwaggerFilePath, filepath.FromSlash("Meta-information/swagger.yaml"))
+		//creating kubernetes configmap with swagger definition
+		fmt.Println("creating configmap with swagger definition")
+		errConf := createConfigMapWithNamespace(swaggerCmName, swaggerPath, flagNamespace, k8sUtils.K8sCreate)
+		if errConf != nil {
+			utils.HandleErrorAndExit("Error creating configmap", errConf)
+		}
+
+	//check if the swagger path is a file
+	case mode.IsRegular():
+		//creating kubernetes configmap with swagger definition
+		fmt.Println("creating configmap with swagger definition")
+		errConf := createConfigMapWithNamespace(swaggerCmName, flagSwaggerFilePath, flagNamespace,
+			k8sUtils.K8sCreate)
+		if errConf != nil {
+			utils.HandleErrorAndExit("Error creating configmap", errConf)
+		}
+	}
+
+	//create API
+	fmt.Println("creating API definition")
+	createAPI(swaggerCmName, nameSuffix)
+}
+
+// validateAddApiCommand validates for required flags and if invalid print error and exit
+func validateAddApiCommand() {
+	// validate mode
+	//configVars := utils.GetMainConfigFromFile(utils.MainConfigFilePath)
+	//if !configVars.Config.KubernetesMode {
+	//	utils.HandleErrorAndExit("set mode to kubernetes with command: apictl set --mode kubernetes",
+	//		errors.New("mode should be set to kubernetes"))
+	//}
+
+	// validate --from-file flag values
+	if _, err := os.Stat(flagSwaggerFilePath); err != nil {
+		utils.HandleErrorAndExit("swagger file path or project not found", err)
+	}
+}
+
+//create configmap with swagger definition
+func createConfigMapWithNamespace(configMapName string, filePath string, namespace string, operation string) error {
```

Comment:
```suggestion
func createConfigMapWithNamespace(configMapName, filePath, namespace, operation string) error {
```

Reply:
You can take out the string since it is common to all the arguments. It improved the readability. Shall we change this like suggested above?

Reply:
Fixed in 2b11a85be6eea5896083c51223ec6860a23d74cd